### PR TITLE
single-pass raft::stats::meanvar

### PR DESCRIPTION
**Pull request by achirkin**
Add a helper to compute column-wise mean and variance for a matrix in a single pass.

Implementation and tests are to a large extent copy-pasted from `raft::stats::mean` and `raft::stats::stddev`.
It is to be used in cuml's `ML::GLM::preProcessData`, where it finishes ~1.8 faster compared to running mean+var separately (on column-major data).

### CODE REVIEW

File: cpp/include/raft/stats/meanvar.hpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#pragma once
+
+#include "detail/meanvar.cuh"
+
+namespace raft::stats {
+
+/**
+ * @brief Compute mean and variance for each column of a given matrix.
+ *
+ * The operation is performed in a single sweep.
```
Comment (cjnolet) [inline cpp/include/raft/stats/meanvar.hpp:26]: To help users who may not immediately see the benefit, can you briefly mention the performance implication / expected speedup of fusing these two operations in a single kernel?

**Review by cjnolet — CHANGES_REQUESTED** [review]
Looks great, just one very minor thing

**Review by cjnolet — APPROVED** [review]
LGTM

**Comment by achirkin** [comment]
rerun tests

**Comment by cjnolet** [comment]
rerun tests

**Comment by cjnolet** [comment]
@gpucibot merge

**Pull request merged**